server/resourcemanager/testutil/test_resource.go

```diff
@@ -70,16 +90,20 @@ func testOperationForContentType(t *testing.T, op operationTester, ct contentTyp
 
 	router := mux.NewRouter()
 	testServer := httptest.NewServer(router)
-	testBridge := rt.RegisterManagerFn(router)
+	manager := rt.RegisterManagerFn(router)
+
+	sortable, ok := manager.Handler().(rm.SortableHandler)
+	if ok {
+		rt.sortFields = sortable.SortingFields()
+	}
 
 	if rt.Prepare != nil {
-		rt.Prepare(t, op.name, testBridge)
+		rt.Prepare(t, op.name, manager)
 	}
 
-	// minor hack to pass "sortFields" to each operation
-	// without needing to propagate sortableHandler for each one
-	sortableHandler := testBridge.(rm.SortableHandler)
-	rt.sortFields = sortableHandler.SortingFields()
+	if !op.needsToRun(manager.EnabledOperations()) {
```

Comment:
this again is a double check to ensure that we are running the tests that we need to run based on the explicit definition and not from infered information. See [this comment](https://github.com/kubeshop/tracetest/pull/2127/files#diff-73567862de3c0b65102c45d3df1f98a49f2649fe588140f55a56fa693068bf51R79)